usaspending_api/broker/management/commands/fpds_nightly_loader.py

```diff
@@ -422,17 +429,22 @@ def add_arguments(self, parser):
             "--id-file",
             metavar="FILEPATH",
             type=str,
-            help="A file containing only transaction IDs (detached_award_procurement_id) "
+            help="(OPTIONAL) A file containing only transaction IDs (detached_award_procurement_id) "
             "to reload, one ID per line. Nonexistent IDs will be ignored.",
         )
 
     @transaction.atomic
     def handle(self, *args, **options):
-        if any([options["detached_award_procurement_ids"], options["id_file"]]):
-            ids_from_file = read_afa_ids_from_file(options["id_file"]) if options["id_file"] else set()
-            explicit_ids = set(options["detached_award_procurement_ids"]) if options["detached_award_procurement_ids"] else set()
-            detached_award_procurement_ids = list(explicit_ids | ids_from_file)
+        if options["id_file"]:
+            self.load_fpds_from_file(options["id_file"])
+
+        if options["detached_award_procurement_ids"]:
+            explicit_ids = options["detached_award_procurement_ids"]
 
-            self.load_specific_transactions(detached_award_procurement_ids)
+            self.load_specific_transactions(explicit_ids)
         else:
             self.nightly_loader(options["date"])
+
+        # Check the linkages from file C to FPDS records and update any that are missing
+        with timer("updating C->D linkages", logger.info):
```

Comment:
The only change by moving this update command here is that it will always run, even if no transactions are loaded/deleted.

Reply:
It also means that it will only run once if load_transactions is called many times, as is now the case when a file is read.

Reply:
I don't know how infrequent there are 0 FPDS transactions in a 24hrs period. What I would like to draw attention to is the possibility of needlessly increasing the time for those runs and when manually testing.

Reply:
Worth looking at. I'll add the "DO NOT MERGE" tag and let this sit at the bottom of the PR list while I figure out if the refactor branch ends up the more practical option.